api/src/controllers/booking.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { Body, Controller, Post } from '@nestjs/common';
+import { BookingService } from '../services/booking.service';
+import { TripSearchDto } from '@shared/dtos/trip-search.dto';
+import { BookTripDto } from '@shared/dtos/book-trip.dto';
+
+@Controller('booking')
+export class BookingController {
+  constructor(private readonly _bookingService: BookingService) {}
+
+  @Post('search')
+  async searchAvailableTrips(@Body() body: TripSearchDto) {
+    const fetchedTrips = await this._bookingService.getAvailableTrips(body);
+
+    if (fetchedTrips?.error != null) {
+      return {
+        response: 'An error occurred whilst fetching available trips.',
+        error: fetchedTrips?.error,
+      };
+    }
+
+    if (fetchedTrips.trips.length > 0) {
+      return {
+        response: 'Trips have been fetched successfully.',
+        data: fetchedTrips.trips,
+      };
+    } else {
+      return { response: 'No trips were found for the given criteria.' };
+    }
+  }
+
+  @Post('book')
```

Comment:
What do you think about changing 'book' to 'reservation' which is a noun?
@Post('reservation')

Reply:
Changed name to reservation.